os/adding-swap.md

```diff
@@ -0,0 +1,90 @@
+# Adding swap in CoreOS/
+Swap is the process of moving pages of memory to a designated part of the hard disk, freeing up space when needed. Swap can be used to alleviate problems with low-memory environments.
+
+By default CoreOS does not include a partition for swap, however one can configure their system to have swap.
+
+## Manually adding a swapfile on ext4
+
+Run these commands **as root** to create a 512 megabyte swapfile.
+
+```
+$ mkdir /var/vm/
+$ fallocate -l 1024m /var/vm/1GiB.swap
+$ chmod 600 /var/vm/1GiB.swap
+$ mkswap /var/vm/1GiB.swap
+$ swapon /var/vm/1GiB.swap
+```
+
+**Note**: The partition on which you are creating the swapfile must meet the following requirements:
+
+* Must not be of type **tmpfs**, **btrfs**, or **xfs**.
+* Must be large enough for your swapfile size.
+
+To verify that our swapfile meets those requirements, run the following command:
+
+```
+$ df -Th
+Filesystem     Type      Size  Used Avail Use% Mounted on
+[...]
+/dev/sdXN      ext4      2.0G  3.0M  1.8G   1% /var
+```
+
+The block device `/dev/sdXN` the correct filesystem type and has enough space for a 1GiB swapfile, so everything should work.
+
+### Create a unit to initialize the swap on boot
+
+The following systemd unit file handles creating, starting, and stopping the swapfile `/etc/systemd/system/swap.service`.
+
+```
+[Unit]
+Description=Turn on swap
+
+[Service]
+Type=oneshot
+Environment="SWAP_PATH=/var/vm" "SWAP_FILE=1GiB.swap"
+ExecStartPre=-/usr/bin/rm -rf ${SWAP_PATH}
+ExecStartPre=/usr/bin/mkdir -p ${SWAP_PATH}
+ExecStartPre=/usr/bin/touch ${SWAP_PATH}/${SWAP_FILE}
+ExecStartPre=/bin/bash -c "fallocate -l 1024m ${SWAP_PATH}/${SWAP_FILE}"
+ExecStartPre=/usr/bin/chmod 600 ${SWAP_PATH}/${SWAP_FILE}
+ExecStartPre=/usr/sbin/mkswap ${SWAP_PATH}/${SWAP_FILE}
+ExecStartPre=/usr/sbin/sysctl vm.swappiness=10
+ExecStart=/sbin/swapon ${SWAP_PATH}/${SWAP_FILE}
+ExecStop=/sbin/swapoff ${SWAP_PATH}/${SWAP_FILE}
+ExecStopPost=-/usr/bin/rm -rf ${SWAP_PATH}
+RemainAfterExit=true
+
+[Install]
+WantedBy=multi-user.target
+```
+
+**Note** Ensure the block device containing your swapfile is configured to auto-mount on boot.
+
+### Enable the unit and start using swap
+
+```
+$ systemctl enable --now /etc/systemd/system/swap.service
```

Comment:
is this equivalent to `systemctl enable /etc/systemd/system/swap.service; systemctl start swap`?